options_compile.cpp

```diff
@@ -227,34 +229,86 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
       fp64Enable = true;
     }
   }
-  effectiveArgs.push_back("-fmodules");
-  if (fp64Enable == 0) {
-    if (szTriple.find("spir64") != szTriple.npos) {
-      if (iCLStdSet <= 120) {
-        effectiveArgs.push_back("-fmodule-file=opencl-c-12-spir64.pcm");
-      } else {
-        effectiveArgs.push_back("-fmodule-file=opencl-c-20-spir64.pcm");
-      }
-    } else if (szTriple.find("spir") != szTriple.npos) {
-      if (iCLStdSet <= 120) {
-        effectiveArgs.push_back("-fmodule-file=opencl-c-12-spir.pcm");
-      } else {
-        effectiveArgs.push_back("-fmodule-file=opencl-c-20-spir.pcm");
+
+  std::map<std::string, bool> extMap {
+    {"cl_khr_3d_image_writes", true},
+    {"cl_khr_depth_images", true},
+    {"cl_khr_fp16", true},
+    {"cl_khr_gl_msaa_sharing", true},
+    {"cl_khr_global_int32_base_atomics", true},
+    {"cl_khr_global_int32_extended_atomics", true},
+    {"cl_khr_int64_base_atomics", true},
+    {"cl_khr_int64_extended_atomics", true},
+    {"cl_khr_local_int32_base_atomics", true},
+    {"cl_khr_local_int32_extended_atomics", true},
+    {"cl_khr_mipmap_image", true},
+    {"cl_khr_mipmap_image_writes", true},
+    {"cl_khr_subgroups", true},
+    {"cl_intel_device_side_avc_motion_estimation", true},
+    {"cl_intel_planar_yuv", true},
+    {"cl_intel_subgroups", true},
+    {"cl_intel_subgroups_short", true}
+  };
+  auto parseClExt = [&](const std::string& clExtStr) {
+    llvm::StringRef clExtRef(clExtStr);
+    clExtRef.consume_front("-cl-ext=");
+    llvm::SmallVector<llvm::StringRef, 32> parsedExt;
+    clExtRef.split(parsedExt, ',');
+    for (llvm::StringRef ext : parsedExt) {
+      char sign = ext.front();
+      bool enabled = sign != '-';
+      llvm::StringRef extName = ext;
+      if (sign == '+' || sign == '-')
+        extName = extName.drop_front();
```

Comment:
```suggestion
      extName = extName.drop_front();
```

I guess we expect particular format here and `+` or `-` is required. I suggest either remove `if` or replace it with an `assert`

Reply:
AFAIK, for Clang's `-cl-ext` option, missing `+` or `-` is legal and equivalent to `+`. I'm following the same logic here.